msdk-featuredetection/msdk-targeteddetection/src/main/java/io/github/msdk/featuredetection/targeteddetection/TargetedDetectionMethod.java

```diff
@@ -164,14 +173,25 @@ public TargetedDetectionMethod(@Nonnull List<IonAnnotation> ionAnnotations,
                 chromatogram.setDataPoints(newDataPoints);
             }
 
+            // Set the m/z value for the chromatogram
+            double[] mzValues = buildingChromatogram.getMzValues();
+            double sum = 0;
+            int values = 0;
+            for(int i=0; i < buildingChromatogram.getSize() ; i++) {
+                if (mzValues[i] > 0) {
+                    values++;
+                    sum = sum + mzValues[i];
+                }
+            }
+            double newMz = sum/values;
+            chromatogram.setMz(newMz);
+
             // Add the chromatogram to the chromatogram list

```

Comment:
That could be true. But it would be good to make a bit of evaluation before doing such change. Also, try discussing this with Matej - I think the original median concept probably came from him.